File: Data Structures & Algorithms/search-for-word/submission-1.java
class Solution {
    private int row;
    private int col;
    private Set<Pair<Integer, Integer>> path = new HashSet<>();

    public boolean exist(char[][] board, String word) {
        row = board.length;
        col = board[0].length;

        for (int r = 0; r < row; r++) {
            for (int c = 0; c < col; c++) {
                if (backtracking(board, word, r, c, 0)) {
                    return true;
                }
            }
        }
        return false;
    }

    private boolean backtracking(char[][] board, String word, int r, int c, int indice) {
        if (indice == word.length()) {
            return true;
        }

        if (r >= row || c >= col || r < 0 || c < 0 || board[r][c] != word.charAt(indice)
            || path.contains(new Pair<>(r, c))) {
            return false;
        }
        if (board[r][c] == word.charAt(indice)) {
            path.add(new Pair<>(r, c));
            boolean res = backtracking(board, word, r + 1, c, indice + 1)
                || backtracking(board, word, r - 1, c, indice + 1)
                || backtracking(board, word, r, c + 1, indice + 1)
                || backtracking(board, word, r, c - 1, indice + 1);
            path.remove(new Pair<>(r, c));
            return res;
        }
        return false;
    }
}

File: Data Structures & Algorithms/valid-sudoku/submission-1.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        for i in range(8):
            s=set()
            for j in range(8):
                if board[i][j]!="." and board[i][j] in s :
                    return False 
                else :
                    s.add(board[i][j])
        
        for j in range(8):
            s=set()
            for i in range(8):
                if board[i][j]!="." and board[i][j] in s :
                    return False 
                else :
                    s.add(board[i][j])
        
        for i in range(3) :
            for j in range(3):
                s=set()
                for l in range(3):
                    for m in range(3):
                        if board[3*i+l][3*j+m]!="." and board[3*i+l][3*j+m] in s :
                            return False 
                        else :
                            s.add(board[3*i+l][3*j+m])
        return True





        